core/test/solver/chebyshev.cpp

```diff
@@ -0,0 +1,375 @@
+            for (gko::size_type j = 0; j < m2->get_size()[1]; ++j) {
+                EXPECT_EQ(m1->at(i, j), m2->at(i, j));
+            }
+        }
+    }
+};
+
+TYPED_TEST_SUITE(Chebyshev, gko::test::ValueTypes, TypenameNameGenerator);
+
+
+TYPED_TEST(Chebyshev, ChebyshevFactoryKnowsItsExecutor)
+{
+    ASSERT_EQ(this->chebyshev_factory->get_executor(), this->exec);
+}
+
+
+TYPED_TEST(Chebyshev, ChebyshevFactoryCreatesCorrectSolver)
+{
+    using Solver = typename TestFixture::Solver;
+    ASSERT_EQ(this->solver->get_size(), gko::dim<2>(3, 3));
+    auto cg_solver = static_cast<Solver*>(this->solver.get());
+    ASSERT_NE(cg_solver->get_system_matrix(), nullptr);
+    ASSERT_EQ(cg_solver->get_system_matrix(), this->mtx);
+}
+
+
+TYPED_TEST(Chebyshev, CanBeCopied)
+{
+    using Mtx = typename TestFixture::Mtx;
+    using Solver = typename TestFixture::Solver;
+    auto copy = this->chebyshev_factory->generate(Mtx::create(this->exec));
+
+    copy->copy_from(this->solver.get());
+
+    ASSERT_EQ(copy->get_size(), gko::dim<2>(3, 3));
+    auto copy_mtx = static_cast<Solver*>(copy.get())->get_system_matrix();
+    this->assert_same_matrices(static_cast<const Mtx*>(copy_mtx.get()),
+                               this->mtx.get());
+}
+
+
+TYPED_TEST(Chebyshev, CanBeMoved)
+{
+    using Mtx = typename TestFixture::Mtx;
+    using Solver = typename TestFixture::Solver;
+    auto copy = this->chebyshev_factory->generate(Mtx::create(this->exec));
+
+    copy->move_from(this->solver);
+
+    ASSERT_EQ(copy->get_size(), gko::dim<2>(3, 3));
+    auto copy_mtx = static_cast<Solver*>(copy.get())->get_system_matrix();
+    this->assert_same_matrices(static_cast<const Mtx*>(copy_mtx.get()),
+                               this->mtx.get());
+}
+
+
+TYPED_TEST(Chebyshev, CanBeCloned)
+{
+    using Mtx = typename TestFixture::Mtx;
+    using Solver = typename TestFixture::Solver;
+    auto clone = this->solver->clone();
+
+    ASSERT_EQ(clone->get_size(), gko::dim<2>(3, 3));
+    auto clone_mtx = static_cast<Solver*>(clone.get())->get_system_matrix();
+    this->assert_same_matrices(static_cast<const Mtx*>(clone_mtx.get()),
+                               this->mtx.get());
+}
+
+
+TYPED_TEST(Chebyshev, CanBeCleared)
+{
+    using Solver = typename TestFixture::Solver;
+    this->solver->clear();
+
+    ASSERT_EQ(this->solver->get_size(), gko::dim<2>(0, 0));
+    auto solver_mtx =
+        static_cast<Solver*>(this->solver.get())->get_system_matrix();
+    ASSERT_EQ(solver_mtx, nullptr);
+}
+
+
+TYPED_TEST(Chebyshev, DefaultApplyUsesInitialGuess)
+{
+    ASSERT_TRUE(this->solver->apply_uses_initial_guess());
+}
+
+
+TYPED_TEST(Chebyshev, CanSetEigenRegion)
+{
+    using Solver = typename TestFixture::Solver;
+    using value_type = typename TestFixture::value_type;
+    std::shared_ptr<Solver> chebyshev_solver =
+        Solver::build()
+            .with_criteria(
+                gko::stop::Iteration::build().with_max_iters(3u).on(this->exec))
```

Comment:
```suggestion
```
not necessary for this test

Reply:
it is not neccessary, but the solver does not allow empty criterion which leads infinite loops.